diff --git a/astropy/wcs/wcs.py b/astropy/wcs/wcs.py
index d3a06f2b3..a023979da 100644
--- a/astropy/wcs/wcs.py
+++ b/astropy/wcs/wcs.py
@@ -1220,6 +1220,10 @@ reduce these to 2 dimensions using the naxis kwarg.
 
             xy = np.hstack([x.reshape((x.size, 1)) for x in axes])
 
+            if xy.shape[0] == 0:
+                return [np.empty(0, dtype=float).reshape(axes[0].shape)
+                        for i in range(self.naxis)]
+
             if ra_dec_order and sky == 'input':
                 xy = self._denormalize_sky(xy)
             output = func(xy, origin)
@@ -1235,6 +1239,8 @@ reduce these to 2 dimensions using the naxis kwarg.
                 raise ValueError(
                     "When providing two arguments, the array must be "
                     "of shape (N, {0})".format(self.naxis))
+            if xy.shape[0] == 0:
+                return np.empty((0, self.naxis), dtype=float)
             if ra_dec_order and sky == 'input':
                 xy = self._denormalize_sky(xy)
             result = func(xy, origin)
