diff --git a/sympy/sets/sets.py b/sympy/sets/sets.py
index 175312287e..200733c260 100644
--- a/sympy/sets/sets.py
+++ b/sympy/sets/sets.py
@@ -59,6 +59,7 @@ class Set(Basic):
     is_UniversalSet = None
     is_Complement = None
     is_ComplexRegion = False
+    is_empty = None
 
     @staticmethod
     def _infimum_key(expr):
@@ -984,6 +985,12 @@ def as_relational(self, x):
     def _measure(self):
         return self.end - self.start
 
+    @property
+    def is_empty(self):
+        if (self.end - self.start).is_positive:
+            return False
+        return None
+
     def to_mpi(self, prec=53):
         return mpi(mpf(self.start._eval_evalf(prec)),
             mpf(self.end._eval_evalf(prec)))
@@ -1468,6 +1475,7 @@ class EmptySet(with_metaclass(Singleton, Set)):
     """
     is_EmptySet = True
     is_FiniteSet = True
+    is_empty = True
 
     @property
     def _measure(self):
