diff --git a/requests/utils.py b/requests/utils.py
index 6a3ed90e..7f6b6441 100644
--- a/requests/utils.py
+++ b/requests/utils.py
@@ -369,11 +369,10 @@ def stream_decode_response_unicode(iterator, r):
     """Stream decodes a iterator."""
 
     if r.encoding is None:
-        for item in iterator:
-            yield item
-        return
+        decoder = codecs.getincrementaldecoder('latin-1')(errors='replace')
+    else:
+        decoder = codecs.getincrementaldecoder(r.encoding)(errors='replace')
 
-    decoder = codecs.getincrementaldecoder(r.encoding)(errors='replace')
     for chunk in iterator:
         rv = decoder.decode(chunk)
         if rv:
