diff --git a/sympy/tensor/array/dense_ndim_array.py b/sympy/tensor/array/dense_ndim_array.py
index 6019daf787..ba75d6e925 100644
--- a/sympy/tensor/array/dense_ndim_array.py
+++ b/sympy/tensor/array/dense_ndim_array.py
@@ -3,8 +3,10 @@
 
 from sympy.core.basic import Basic
 from sympy.core.containers import Tuple
+from sympy.core.numbers import Integer
 from sympy.core.singleton import S
 from sympy.core.sympify import _sympify
+from sympy.external.gmpy import SYMPY_INTS
 from sympy.tensor.array.mutable_ndim_array import MutableNDimArray
 from sympy.tensor.array.ndim_array import NDimArray, ImmutableNDimArray, ArrayKind
 from sympy.utilities.iterables import flatten
@@ -196,6 +198,8 @@ def __setitem__(self, index, value):
                 other_i = [ind - j for ind, j in zip(i, slice_offsets) if j is not None]
                 self._array[self._parse_index(i)] = value[other_i]
         else:
+            if isinstance(index, (SYMPY_INTS, Integer)):
+                index = (index,)
             index = self._parse_index(index)
             self._setter_iterable_check(value)
             value = _sympify(value)
