diff --git a/.astropy-root b/.astropy-root
deleted file mode 100644
index e69de29bb..000000000
diff --git a/astropy/__init__.py b/astropy/__init__.py
index 269aa780d..dbad08817 100644
--- a/astropy/__init__.py
+++ b/astropy/__init__.py
@@ -148,10 +148,9 @@ def _initialize_astropy():
         # not installed
         if (Path(__file__).parent.parent / ".astropy-root").exists():
             raise ImportError(
-                "You appear to be trying to import astropy from "
-                "within a source checkout or from an editable "
-                "installation without building the extension "
-                "modules first. Either run:\n\n"
+                "You appear to be trying to import astropy from within a source checkout "
+                "or from an editable installation without building the extension modules "
+                "first. Either run:\n\n"
                 "  pip install -e .\n\nor\n\n"
                 "  python setup.py build_ext --inplace\n\n"
                 "to make sure the extension modules are built "
