diff --git a/lib/matplotlib/cbook.py b/lib/matplotlib/cbook.py
index 483039e337..5d75bca12e 100644
--- a/lib/matplotlib/cbook.py
+++ b/lib/matplotlib/cbook.py
@@ -1657,7 +1657,10 @@ def _safe_first_finite(obj, *, skip_nonfinite=True):
         raise RuntimeError("matplotlib does not "
                            "support generators as input")
     else:
-        return next(val for val in obj if safe_isfinite(val))
+        try:
+            return next(val for val in obj if safe_isfinite(val))
+        except StopIteration:
+            return None
 
 
 def sanitize_sequence(data):
