diff --git a/lib/matplotlib/widgets.py b/lib/matplotlib/widgets.py
index ed130e6854..3feae95b10 100644
--- a/lib/matplotlib/widgets.py
+++ b/lib/matplotlib/widgets.py
@@ -134,6 +134,17 @@ class AxesWidget(Widget):
         self.canvas = ax.figure.canvas
         self._cids = []
 
+    def __getstate__(self):
+        # The canvas cannot be pickled (e.g., if it is a Qt canvas), but it
+        # can be recovered from self.ax.figure.canvas after unpickling.
+        state = self.__dict__.copy()
+        del state['canvas']
+        return state
+
+    def __setstate__(self, state):
+        self.__dict__.update(state)
+        self.canvas = self.ax.figure.canvas
+
     def connect_event(self, event, callback):
         """
         Connect a callback function with an event.
