diff --git a/django/db/migrations/autodetector.py b/django/db/migrations/autodetector.py
index bf9a45530a..e2ddb468d1 100644
--- a/django/db/migrations/autodetector.py
+++ b/django/db/migrations/autodetector.py
@@ -1097,6 +1097,16 @@ class MigrationAutodetector:
                         field = self.new_apps.get_model(app_label, model_name)._meta.get_field(field_name)
                         if field.remote_field and field.remote_field.model:
                             dependencies.extend(self._get_dependencies_for_foreign_key(field))
+                # Ensure any fields being removed that are part of the old
+                # unique_together/index_together are removed after this operation.
+                for old_foo_together in old_value:
+                    for field_name in old_foo_together:
+                        if (app_label, model_name, field_name) in (
+                            self.old_field_keys - self.new_field_keys
+                        ):
+                            dependencies.append(
+                                (app_label, model_name, field_name, False)
+                            )
 
                 self.add_operation(
                     app_label,
