diff --git a/django/urls/resolvers.py b/django/urls/resolvers.py
index 0912a82f14..14da2f9cad 100644
--- a/django/urls/resolvers.py
+++ b/django/urls/resolvers.py
@@ -435,7 +435,7 @@ class URLResolver:
                 ).format(status_code=status_code, path=path)
                 messages.append(Error(msg, hint=str(e), id='urls.E008'))
                 continue
-            signature = inspect.signature(handler)
+            signature = inspect.signature(handler, follow_wrapped=False)
             args = [None] * num_parameters
             try:
                 signature.bind(*args)
diff --git a/tests/check_framework/test_urls.py b/tests/check_framework/test_urls.py
index f0c257bc77..201dee4ba7 100644
--- a/tests/check_framework/test_urls.py
+++ b/tests/check_framework/test_urls.py
@@ -209,6 +209,11 @@ class CheckCustomErrorHandlersTests(SimpleTestCase):
         result = check_url_config(None)
         self.assertEqual(result, [])
 
+    @override_settings(ROOT_URLCONF='check_framework.urls.good_error_handlers_cbv')
+    def test_good_handlers_class_based_view(self):
+        result = check_url_config(None)
+        self.assertEqual(result, [])
+
 
 class CheckURLSettingsTests(SimpleTestCase):
 
diff --git a/tests/check_framework/urls/good_error_handlers_cbv.py b/tests/check_framework/urls/good_error_handlers_cbv.py
new file mode 100644
index 0000000000..14c0ee5db0
--- /dev/null
+++ b/tests/check_framework/urls/good_error_handlers_cbv.py
@@ -0,0 +1,14 @@
+from django.views.generic import View
+
+urlpatterns = []
+
+
+class GoodHandlerCBV(View):
+    def get(self, request, exception=None):
+        pass
+
+
+handler400 = GoodHandlerCBV.as_view()
+handler403 = GoodHandlerCBV.as_view()
+handler404 = GoodHandlerCBV.as_view()
+handler500 = GoodHandlerCBV.as_view()
