diff --git a/sympy/printing/latex.py b/sympy/printing/latex.py
index fc9e61b3b1..a8f42a9e0e 100644
--- a/sympy/printing/latex.py
+++ b/sympy/printing/latex.py
@@ -851,7 +851,8 @@ def _print_Function(self, expr, exp=None):
                 len(args) == 1 and \
                 not self._needs_function_brackets(expr.args[0])
 
-            inv_trig_table = ["asin", "acos", "atan", "acsc", "asec", "acot"]
+            inv_trig_table = ["asin", "acos", "atan", "acsc", "asec", "acot",
+                              "asinh", "acosh", "atanh", "acsch", "asech", "acoth"]
 
             # If the function is an inverse trig function, handle the style
             if func in inv_trig_table:
