diff --git a/sympy/tensor/array/dense_ndim_array.py b/sympy/tensor/array/dense_ndim_array.py
index 6019daf787..ca443889af 100644
--- a/sympy/tensor/array/dense_ndim_array.py
+++ b/sympy/tensor/array/dense_ndim_array.py
@@ -196,6 +196,8 @@ def __setitem__(self, index, value):
                 other_i = [ind - j for ind, j in zip(i, slice_offsets) if j is not None]
                 self._array[self._parse_index(i)] = value[other_i]
         else:
+            if not isinstance(index, tuple):
+                index = (index,)
             index = self._parse_index(index)
             self._setter_iterable_check(value)
             value = _sympify(value)
