diff --git a/sympy/utilities/codegen.py b/sympy/utilities/codegen.py
index f0befb2bd..07508f529 100644
--- a/sympy/utilities/codegen.py
+++ b/sympy/utilities/codegen.py
@@ -739,7 +739,14 @@ def routine(self, name, expr, argument_sequence=None, global_vars=None):
                 try:
                     new_args.append(name_arg_dict[symbol])
                 except KeyError:
-                    new_args.append(InputArgument(symbol))
+                    # symbol not in expression, check if it's a MatrixSymbol
+                    # and add appropriate array metadata
+                    if isinstance(symbol, MatrixSymbol):
+                        dims = tuple([(S.Zero, dim - 1) for dim in symbol.shape])
+                        metadata = {'dimensions': dims}
+                    else:
+                        metadata = {}
+                    new_args.append(InputArgument(symbol, **metadata))
             arg_list = new_args
 
         return Routine(name, arg_list, return_val, local_vars, global_vars)
