diff --git a/sympy/tensor/array/dense_ndim_array.py b/sympy/tensor/array/dense_ndim_array.py
index 6019daf787..493b1a9ec3 100644
--- a/sympy/tensor/array/dense_ndim_array.py
+++ b/sympy/tensor/array/dense_ndim_array.py
@@ -5,6 +5,8 @@
 from sympy.core.containers import Tuple
 from sympy.core.singleton import S
 from sympy.core.sympify import _sympify
+from sympy.external.gmpy import SYMPY_INTS
+from sympy.core.numbers import Integer
 from sympy.tensor.array.mutable_ndim_array import MutableNDimArray
 from sympy.tensor.array.ndim_array import NDimArray, ImmutableNDimArray, ArrayKind
 from sympy.utilities.iterables import flatten
@@ -190,6 +192,9 @@ def __setitem__(self, index, value):
         [[1, 0], [0, 1]]
 
         """
+        if isinstance(index, (SYMPY_INTS, Integer, slice)):
+            index = (index,)
+
         if isinstance(index, tuple) and any(isinstance(i, slice) for i in index):
             value, eindices, slice_offsets = self._get_slice_data_for_array_assignment(index, value)
             for i in eindices:
