diff --git a/sympy/core/sympify.py b/sympy/core/sympify.py
index e04b8112d1..21fc569709 100644
--- a/sympy/core/sympify.py
+++ b/sympy/core/sympify.py
@@ -3,11 +3,13 @@
 from __future__ import print_function, division
 
 from typing import Dict, Type, Callable, Any
+import warnings
 
 from inspect import getmro
 
 from .compatibility import iterable
 from .parameters import global_parameters
+from ..utilities.exceptions import SymPyDeprecationWarning
 
 
 class SympifyError(ValueError):
@@ -384,6 +386,13 @@ def sympify(a, locals=None, convert_xor=True, strict=False, rational=False,
     # As a last chance, we try to take "a"'s normal form via unicode()
     # and try to parse it. If it fails, then we have no luck and
     # return an exception
+    SymPyDeprecationWarning(
+        "sympifying a string or custom object with __str__ or __repr__ "
+        "is deprecated",
+        issue=19006,
+        deprecated_since_version="1.6",
+        stacklevel=3,
+    ).warn()
     try:
         from .compatibility import unicode
         a = unicode(a)
