diff --git a/sympy/tensor/array/ndim_array.py b/sympy/tensor/array/ndim_array.py
index 6490a655a4..6c03b7acbf 100644
--- a/sympy/tensor/array/ndim_array.py
+++ b/sympy/tensor/array/ndim_array.py
@@ -145,7 +145,10 @@ def __new__(cls, iterable, shape=None, **kwargs):
 
     def _parse_index(self, index):
         if isinstance(index, (SYMPY_INTS, Integer)):
-            raise ValueError("Only a tuple index is accepted")
+            if self._rank == 1:
+                index = (index,)
+            else:
+                raise ValueError("Only a tuple index is accepted")
 
         if self._loop_size == 0:
             raise ValueError("Index not valide with an empty array")
