diff --git a/django/forms/formsets.py b/django/forms/formsets.py
index eafe35720..02f89eb5d 100644
--- a/django/forms/formsets.py
+++ b/django/forms/formsets.py
@@ -178,12 +178,19 @@ class BaseFormSet(object):
         """Return a list of all the extra forms in this formset."""
         return self.forms[self.initial_form_count():]
 
+    def _get_empty_form_kwargs(self):
+        """
+        Return additional keyword arguments to include with the empty form.
+        """
+        return {}
+
     @property
     def empty_form(self):
         form = self.form(
             auto_id=self.auto_id,
             prefix=self.add_prefix('__prefix__'),
             empty_permitted=True,
+            **self._get_empty_form_kwargs()
         )
         self.add_fields(form, None)
         return form
