diff --git a/sympy/tensor/array/dense_ndim_array.py b/sympy/tensor/array/dense_ndim_array.py
index 6019daf787..5ca23744dc 100644
--- a/sympy/tensor/array/dense_ndim_array.py
+++ b/sympy/tensor/array/dense_ndim_array.py
@@ -196,6 +196,7 @@ def __setitem__(self, index, value):
                 other_i = [ind - j for ind, j in zip(i, slice_offsets) if j is not None]
                 self._array[self._parse_index(i)] = value[other_i]
         else:
+            index = self._check_index_for_getitem(index)
             index = self._parse_index(index)
             self._setter_iterable_check(value)
             value = _sympify(value)
diff --git a/sympy/tensor/array/sparse_ndim_array.py b/sympy/tensor/array/sparse_ndim_array.py
index f11aa95be8..effb91110c 100644
--- a/sympy/tensor/array/sparse_ndim_array.py
+++ b/sympy/tensor/array/sparse_ndim_array.py
@@ -181,6 +181,7 @@ def __setitem__(self, index, value):
                 elif complete_index in self._sparse_array:
                     self._sparse_array.pop(complete_index)
         else:
+            index = self._check_index_for_getitem(index)
             index = self._parse_index(index)
             value = _sympify(value)
             if value == 0 and index in self._sparse_array:
