diff --git a/requests/utils.py b/requests/utils.py
index 6a3ed90e..869ce337 100644
--- a/requests/utils.py
+++ b/requests/utils.py
@@ -369,11 +369,15 @@ def stream_decode_response_unicode(iterator, r):
     """Stream decodes a iterator."""
 
     if r.encoding is None:
-        for item in iterator:
-            yield item
-        return
+        encoding = r.apparent_encoding
+        if encoding is None:
+            for item in iterator:
+                yield item
+            return
+    else:
+        encoding = r.encoding
 
-    decoder = codecs.getincrementaldecoder(r.encoding)(errors='replace')
+    decoder = codecs.getincrementaldecoder(encoding)(errors='replace')
     for chunk in iterator:
         rv = decoder.decode(chunk)
         if rv:
