diff --git a/django/db/models/base.py b/django/db/models/base.py
index 0b8425aa85..aed5a34ec5 100644
--- a/django/db/models/base.py
+++ b/django/db/models/base.py
@@ -521,14 +521,14 @@ class Model(metaclass=ModelBase):
         return '%s object (%s)' % (self.__class__.__name__, self.pk)
 
     def __eq__(self, other):
-        if not isinstance(other, Model):
-            return False
-        if self._meta.concrete_model != other._meta.concrete_model:
-            return False
-        my_pk = self.pk
-        if my_pk is None:
-            return self is other
-        return my_pk == other.pk
+        if isinstance(other, Model):
+            if self._meta.concrete_model != other._meta.concrete_model:
+                return False
+            my_pk = self.pk
+            if my_pk is None:
+                return self is other
+            return my_pk == other.pk
+        return NotImplemented
 
     def __hash__(self):
         if self.pk is None:
