diff --git a/lib/matplotlib/colors.py b/lib/matplotlib/colors.py
index 05922c93df..5223252e44 100644
--- a/lib/matplotlib/colors.py
+++ b/lib/matplotlib/colors.py
@@ -48,6 +48,7 @@ import io
 import itertools
 from numbers import Real
 import re
+import warnings
 
 from PIL import Image
 from PIL.PngImagePlugin import PngInfo
@@ -1307,7 +1308,10 @@ class Normalize:
         is_scalar = not np.iterable(value)
         if is_scalar:
             value = [value]
-        dtype = np.min_scalar_type(value)
+        with warnings.catch_warnings():
+            warnings.filterwarnings("ignore",
+                                    "converting a masked element to nan.")
+            dtype = np.min_scalar_type(value)
         if np.issubdtype(dtype, np.integer) or dtype.type is np.bool_:
             # bool_/int8/int16 -> float32; int32/int64 -> float64
             dtype = np.promote_types(dtype, np.float32)
