diff --git a/sympy/printing/octave.py b/sympy/printing/octave.py
index 9fa66d4d7..33377528a 100644
--- a/sympy/printing/octave.py
+++ b/sympy/printing/octave.py
@@ -392,6 +392,10 @@ def _print_lowergamma(self, expr):
                                               self._print(expr.args[0]))
 
 
+    def _print_sinc(self, expr):
+        return "sinc(%s/pi)" % self._print(expr.args[0])
+
+
     def _print_hankel1(self, expr):
         return "besselh(%s, 1, %s)" % (self._print(expr.order),
                                        self._print(expr.argument))
diff --git a/sympy/printing/tests/test_octave.py b/sympy/printing/tests/test_octave.py
index 0907625a8..1cf58a995 100644
--- a/sympy/printing/tests/test_octave.py
+++ b/sympy/printing/tests/test_octave.py
@@ -10,6 +10,7 @@
                                             besselk, hankel1, hankel2, airyai,
                                             airybi, airyaiprime, airybiprime)
 from sympy.functions.special.gamma_functions import (lowergamma, uppergamma)
+from sympy.functions.elementary.trigonometric import sinc
 from sympy.utilities.pytest import XFAIL
 from sympy.core.compatibility import range
 
@@ -368,3 +369,8 @@ def test_specfun():
     assert octave_code(lowergamma(n, x)) == 'gammainc(x, n, \'lower\')'
     assert octave_code(jn(n, x)) == 'sqrt(2)*sqrt(pi)*sqrt(1./x).*besselj(n + 1/2, x)/2'
     assert octave_code(yn(n, x)) == 'sqrt(2)*sqrt(pi)*sqrt(1./x).*bessely(n + 1/2, x)/2'
+
+
+def test_sinc():
+    assert octave_code(sinc(x)) == 'sinc(x/pi)'
+    assert octave_code(sinc(2*x)) == 'sinc(2*x/pi)'
