diff --git a/lib/matplotlib/colors.py b/lib/matplotlib/colors.py
index 05922c93df..b561da1c5d 100644
--- a/lib/matplotlib/colors.py
+++ b/lib/matplotlib/colors.py
@@ -48,6 +48,7 @@ import io
 import itertools
 from numbers import Real
 import re
+import warnings
 
 from PIL import Image
 from PIL.PngImagePlugin import PngInfo
@@ -1307,7 +1308,12 @@ class Normalize:
         is_scalar = not np.iterable(value)
         if is_scalar:
             value = [value]
-        dtype = np.min_scalar_type(value)
+        # Suppress the warning when value is a masked array with all masked
+        # values, as np.min_scalar_type will try to convert the masked values
+        # to a scalar, which triggers a numpy warning.
+        with warnings.catch_warnings():
+            warnings.simplefilter('ignore')
+            dtype = np.min_scalar_type(value)
         if np.issubdtype(dtype, np.integer) or dtype.type is np.bool_:
             # bool_/int8/int16 -> float32; int32/int64 -> float64
             dtype = np.promote_types(dtype, np.float32)
