diff --git a/sympy/matrices/expressions/matmul.py b/sympy/matrices/expressions/matmul.py
index f5b95ee95..cd653ea0d 100644
--- a/sympy/matrices/expressions/matmul.py
+++ b/sympy/matrices/expressions/matmul.py
@@ -64,7 +64,7 @@ def _entry(self, i, j, expand=True):
         if X.has(ImmutableMatrix) or Y.has(ImmutableMatrix):
             return coeff*Add(*[X[i, k]*Y[k, j] for k in range(X.cols)])
         result = Sum(coeff*X[i, k]*Y[k, j], (k, 0, X.cols - 1))
-        if not X.cols.is_number:
+        if hasattr(X.cols, 'is_number') and X.cols.is_number is False:
             # Don't waste time in result.doit() if the sum bounds are symbolic
             expand = False
         return result.doit() if expand else result
