diff --git a/sympy/core/sympify.py b/sympy/core/sympify.py
index e04b8112d1..131e5df184 100644
--- a/sympy/core/sympify.py
+++ b/sympy/core/sympify.py
@@ -384,6 +384,17 @@ def sympify(a, locals=None, convert_xor=True, strict=False, rational=False,
     # As a last chance, we try to take "a"'s normal form via unicode()
     # and try to parse it. If it fails, then we have no luck and
     # return an exception
+    if not isinstance(a, str):
+        from sympy.utilities.exceptions import SymPyDeprecationWarning
+        SymPyDeprecationWarning(
+            feature="Implicit conversion of objects `__str__`/`__repr__` to a SymPy "
+                    "expression is deprecated. This behavior is deprecated since SymPy "
+                    "1.6. See this issue for more information, especially the workaround "
+                    "for the current behavior:\n"
+                    "https://github.com/sympy/sympy/issues/18066",
+            deprecated_since_version="1.6",
+            issue=18066,
+        ).warn(stacklevel=3)
     try:
         from .compatibility import unicode
         a = unicode(a)
