diff --git a/django/db/models/options.py b/django/db/models/options.py
index be3d68f390..9b60102b92 100644
--- a/django/db/models/options.py
+++ b/django/db/models/options.py
@@ -206,7 +206,7 @@ class Options:
 
             self.unique_together = normalize_together(self.unique_together)
             self.index_together = normalize_together(self.index_together)
-            if self.index_together:
+            if self.index_together and not self.indexes:
                 warnings.warn(
                     f"'index_together' is deprecated. Use 'Meta.indexes' in "
                     f"{self.label!r} instead.",
