diff --git a/django/db/models/query_utils.py b/django/db/models/query_utils.py
index 27430ae4ca..7f8f110426 100644
--- a/django/db/models/query_utils.py
+++ b/django/db/models/query_utils.py
@@ -5,6 +5,7 @@ Factored out from django.db.models.query to avoid making the main module very
 large and/or so that they can be used by other modules without getting into
 circular import difficulties.
 """
+import copy
 import functools
 import inspect
 from collections import namedtuple
@@ -61,6 +62,12 @@ class Q(tree.Node):
     def _combine(self, other, conn):
         if not isinstance(other, Q):
             raise TypeError(other)
+        # If the other Q() is empty, ignore it and just use `self`.
+        if not other:
+            return copy.deepcopy(self)
+        # Or if this Q is empty, ignore it and just use `other`.
+        if not self:
+            return copy.deepcopy(other)
         obj = type(self)()
         obj.connector = conn
         obj.add(self, conn)
diff --git a/tests/queries/test_q.py b/tests/queries/test_q.py
index 9b9fba6ba1..27e519e7b7 100644
--- a/tests/queries/test_q.py
+++ b/tests/queries/test_q.py
@@ -74,3 +74,27 @@ class QTests(SimpleTestCase):
         q = q1 & q2
         path, args, kwargs = q.deconstruct()
         self.assertEqual(Q(*args, **kwargs), q)
+
+    def test_combine_and_empty(self):
+        q = Q(x=1)
+        self.assertEqual(q & Q(), q)
+        self.assertEqual(Q() & q, q)
+
+    def test_combine_and_both_empty(self):
+        self.assertEqual(Q() & Q(), Q())
+
+    def test_combine_or_empty(self):
+        q = Q(x=1)
+        self.assertEqual(q | Q(), q)
+        self.assertEqual(Q() | q, q)
+
+    def test_combine_or_both_empty(self):
+        self.assertEqual(Q() | Q(), Q())
+
+    def test_combine_not_q_object(self):
+        obj = object()
+        q = Q(x=1)
+        with self.assertRaisesMessage(TypeError, str(obj)):
+            q | obj
+        with self.assertRaisesMessage(TypeError, str(obj)):
+            q & obj
